src/core/OffsetNode.h

```diff
@@ -1,21 +1,31 @@
 #pragma once
 
-#include "core/node.h"
+#include <memory>
+#include <string>
+
 #include "core/ModuleInstantiation.h"
+#include "core/node.h"
 #include "clipper2/clipper.h"
 
-#include <string>
+class CurveDiscretizer;
 
 class OffsetNode : public AbstractPolyNode
 {
 public:
   VISITABLE();
-  OffsetNode(const ModuleInstantiation *mi) : AbstractPolyNode(mi) {}
+  OffsetNode(const ModuleInstantiation *mi, std::shared_ptr<CurveDiscretizer> discretizer)
+    : AbstractPolyNode(mi), discretizer(std::move(discretizer))
+  {
+  }
+
+  ~OffsetNode();
```

Comment:
I think this is an artifact of me testing with reference type and can be removed.